config/dependencies/gateway-api/gateway/kustomization.yaml

```
@@ -0,0 +1,5 @@
+---
+# Adds namespace to all resources.
+namespace: istio-system
+resources:
+- gateway.yaml
```

Have you considered the following?

```suggestion
- github.com/kubernetes-sigs/gateway-api/config/crd?ref=master
```